### PR TITLE
Reorder dimensions

### LINKED ISSUE #236: Reorder dimensions to start with space and time
**Is your feature request related to a problem? Please describe.**
In our current [dataset structure](http://127.0.0.1:3000/docs/build/html/getting_started/movement_dataset.html) the order of `xarray` dimensions is: `time`, `individuals`, `keypoints`, `space`.

This is not very intuitive, namely because `time` and `space` are the most essential dimensions which should always exist regardless of how motion was tracked. As we move toward supporting more types of data, and possibly making non-spacetime dimensions optional (e.g. create `keypoints` only if more than 1 is present, and ditto for `individuals`), and it makes more sense to put the "essential" dimensions at the beginning and the optional ones at the end.

This was brought up by @sfmig, and we both think this would make sense.

**Describe the solution you'd like**
Reorder dimensions in order of "essentiality": `time`, `space`, `keypoints`,  `individuals`, `views` (if we adopt #196).
This shouldn't be too hard as we could simply add xarray's [`transpose()`](https://docs.xarray.dev/en/stable/user-guide/reshaping.html#reordering-dimensions) methods at the end of our loading functions to get the desired order. But we would have to adapt validators (related to #210), tests and docs.

**Describe alternatives you've considered**
- `space`, `time`, `keypoints`, `individual`, `views`
- Keep things as they are, don't fix it if it ain't broken?

**Additional context**
- [Discussions](https://neuroinformatics.zulipchat.com/#narrow/stream/406001-Movement/topic/Support.20single.20individual.20data.20without.20the.20.60individuals.60.20dim) with @vigji on essential vs optional dimensions.
- Support for multiple views [issue](https://github.com/neuroinformatics-unit/movement/issues/196) and PR #197 
- On re-designing the validators: #210

### PR DESCRIPTION
**Pull request by lochhh**
## Description

**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [x] Other

**Why is this PR needed?**
This PR closes #236. 

**What does this PR do?**
This PR 
- reorders the dimensions of movement `poses` and `bboxes` datasets as follows:
  - `poses`: `("time", "individuals", "keypoints", "space")` &rarr; `("time", "space", "keypoints", "individuals")`
  - `bboxes`:  `("time", "individuals", "space")` &rarr; `("time", "space", "individuals")`
- updates the docs (by first checking changed files, and then searching for the keyword "shape" in the entire directory) 
- removes the duplicated `assert_dataset` in `test_load_poses` and `test_load_bboxes` by introducing a new helper class `MovementDatasetAsserts` in `conftest.py` that provides a common `valid_dataset` function to assert poses and bboxes dataset validity in tests
- replaces if-else-blocks with a ternary operator
- fixes broken examples (in docstrings and in .md)
- fixes broken references to external packages

**Strategy**:
- [x] Directly transpose after datasets are loaded as xarray.Datasets (i.e. validators are still using the old "dims order") 
- [x] Make tests pass (by transposing fixtures) for both poses and bboxes
- [x] Change the order in Validators (at the root) so that we don't need to transpose the dataset after loading them into movement
- [x] Drop transpose in test fixtures and make tests pass by creating fixtures with the new "dims order"
- [x] Refactor tests (e.g. [duplicated code](https://github.com/neuroinformatics-unit/movement/pull/351/checks?check_run_id=33561095934) )
- [x] Update documentation/docstrings

## References
#236

## How has this PR been tested?
Tests pass locally and on CI.

## Does this PR require an update to the documentation?
Done.

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Review by niksirbi — APPROVED** [review]
Thanks for undertaking this tricky surgery @lochhh, a PR that touches almost all files!

I couldn't find a fault with it and I'd say this is good to go, after resolving merge conflicts.

There is one thing I'm puzzled about, which you perhaps can help with.
In `compute_norm()`, we have:

```python
if "space" in data.dims:
        validate_dims_coords(data, {"space": ["x", "y"]})
        return xr.apply_ufunc(
            np.linalg.norm,
            data,
            input_core_dims=[["space"]],
            kwargs={"axis": -1},
        )
 ```
 
 `kwargs={"axis": -1}` is passed onto `linalg.norm` and I understood it as meaning the "norm along the space axis". You haven't updated this, so by my logic this should be broken now, but the passing tests indicate that it works as expected. What gives?

**Comment by lochhh** [comment]
> Thanks for undertaking this tricky surgery @lochhh, a PR that touches almost all files!
> 
> I couldn't find a fault with it and I'd say this is good to go, after resolving merge conflicts.
> 
> There is one thing I'm puzzled about, which you perhaps can help with. In `compute_norm()`, we have:
> 
> ```python
> if "space" in data.dims:
>         validate_dims_coords(data, {"space": ["x", "y"]})
>         return xr.apply_ufunc(
>             np.linalg.norm,
>             data,
>             input_core_dims=[["space"]],
>             kwargs={"axis": -1},
>         )
> ```
> 
> `kwargs={"axis": -1}` is passed onto `linalg.norm` and I understood it as meaning the "norm along the space axis". You haven't updated this, so by my logic this should be broken now, but the passing tests indicate that it works as expected. What gives?

From [xarray's `apply_ufunc` tutorial](https://tutorial.xarray.dev/advanced/apply_ufunc/core-dimensions.html):
> With `apply_ufunc`, core dimensions are recognized by name, and then moved to the last dimension of any input arguments before applying the given function. This means that for functions that accept an axis argument, you usually need to set axis=-1

**Comment by niksirbi** [comment]
> With `apply_ufunc`, core dimensions are recognized by name, and then moved to the last dimension of any input arguments before applying the given function. This means that for functions that accept an axis argument, you usually need to set axis=-1

That's cool, thanks for clarifying! LGTM 🎉

**Pull request merged**
